feature-libs/cart/quick-order/core/services/quick-order.service.spec.ts

```diff
@@ -325,24 +325,26 @@ describe('QuickOrderService', () => {
       });
   });
 
-  it('should add deleted entry and after 7s delete it', (done) => {
+  it('should add deleted entry and after 7s delete it', fakeAsync(() => {
     service.loadEntries(mockEntries);
     service.softDeleteEntry(0);
 
     service
       .getSoftDeletedEntries()
-      .pipe(
-        tap((softDeletedEntries) => {
-          expect(softDeletedEntries).toEqual({ mockCode1: mockEntry1 });
-        }),
-        delay(7000),
-        switchMap(() => service.getSoftDeletedEntries())
-      )
+      .pipe(take(1))
+      .subscribe((result) => {
+        expect(result).toEqual({ mockCode1: mockEntry1 });
+      });
+
+    tick(7000);
+
+    service
+      .getSoftDeletedEntries()
+      .pipe(take(1))
       .subscribe((result) => {
         expect(result).toEqual({});
       });
-    done();
-  });
+  }));
```

Comment:
[MINOR]
I can see `expects` in two `.subscribe()` callbacks. How are we sure those `expects` will be called, if the implementation is possibly broken in the future so the observable doesn't emit at all?

I believe we should use either:
- call `done()` callback in the subscription callback (however now we have 2 subscribe callbacks. And we shouldn't call `done()` more than once!)
- if the observable is meant to emit synchronously, we can simply copy the `result` to a variable declared higher above. And then **after the subscription** call the `expect(variable).toEqual(xxx)`